client/reader/discover/discover-stream.js

```diff
@@ -143,9 +187,12 @@ const DiscoverStream = ( props ) => {
 	}
 
 	return (
-		<Stream { ...streamProps }>
-			<HeaderAndNavigation />
-		</Stream>
+		<>
+			<Stream { ...streamProps }>
+				<HeaderAndNavigation />
+			</Stream>
+			{ ! isLoggedIn && <ReaderDiscoverLoginPrompt /> }
```

Comment:
Maybe we should move this component inside stream so that it can be used easily by any other stream 🤔

Reply:
Yeah, I think so. Let's give that a try. Done in c53eaf27bdd85129ecc7ff7e88d5ffacabd701b2